File: luogu/03_循环结构/P_1217.cpp
#include<iostream> 
#include<algorithm>
#include<cmath>
using namespace std;
int main(){
	int a, b;
	cin >> a >> b;
	for( ; a <= b; a++){
		int i;
		int temp = a;
		int reverse = 0;
		while(temp > 0){
			reverse = reverse * 10 + temp % 10;
			temp = temp / 10;
		} 
		if (a != reverse){
			continue;
		}
		int sqrt_a = sqrt(double(a));
		for (i = 2; i <= sqrt_a; i++){
			if (a % i == 0){
				break;
			}
		}
		if (i > sqrt_a){
			cout << a <<endl;
		}
	}
}

File: luogu/03_循环结构/P_1217.py
import math
a, b = map(int, input().split())
while a <= b:
    sqrt_a = math.sqrt(a)
    i = 2
    while i <= sqrt_a:
        if a % i == 0:
            break
        i += 1
    if i > sqrt_a:
        str_a = str(a)
        if str_a == str_a[-1::-1]:
            print(a)
    a += 1
